andela/travel_tool_front
# #161646967 Remove red asterisks on optional fields on the checklist form

#### What does this PR do?
Removes the red asterisks that signalled and mislead users that certain optional fields are required.

#### Description of Task to be completed?
 - Add code to check whether the field is explicitly set as optional.
 - Fix styling issues that arise from removing the said asterisks.

#### How should this be manually tested?
  - Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Switch to the project directory
   `cd travel_tool_front`
 - Checkout the **bg-fix-required-fields-on-checklist-form-161646967** branch
   `git checkout bg-fix-required-fields-on-checklist-form-161646967`
 - Install dependencies and start the server
   `make start`
 - Start the [backend](https://github.com/travel_tool_back) of this project.
 - Navigate to [travela](http://travela-local.andela.com:3000/) in your browser and log in with your Andela account. 
- On your database, modify your **roleId** in the **UserRoles** table to **10948** (Super Admin).
- Navigate to the Trip Planner tab and click on **Add Item**. You should notice that the misleading asterisks have been removed from optional fields. You should be able to create a checklist item normally.

#### Any background context you want to provide?
Ignore the .snap files in the files changed section.
The Forms API assumes that all inputs are required and therefore a red asterisk is displayed. However, this is not enforced with HTML.

#### What are the relevant pivotal tracker stories?
[#161646967](https://www.pivotaltracker.com/story/show/161646967)

#### Screenshots
![image](https://user-images.githubusercontent.com/41262970/47869241-7a2a3280-de17-11e8-8b57-7279f5f4fa55.png)
